src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ReferenceCountedDisposable.cs

```diff
@@ -51,7 +53,7 @@ namespace Roslyn.Utilities
     /// even in concurrent execution.</para>
     /// </remarks>
     /// <typeparam name="T">The type of disposable object.</typeparam>
-    internal sealed class ReferenceCountedDisposable<T> : IReferenceCountedDisposable<T>, IDisposable
+    internal sealed class ReferenceCountedDisposable<T> : IReferenceCountedDisposable<T>
         where T : class, IDisposable
```

Comment:
i examined making `T` constrained on IAsyncDisposable as well.  However, this didn't work for all cases of how we use this type.  For examlpe, we make a `ReferenceCountedDisposable<MemoryMappedFile>`.  That inner type is only IDispoable.  So, instead, we require that the tyep be at least IDIsposable, but we respect and forward along IAsyncDisposable if possible.

Reply:
This is fine. The alternative would be cloning this to create a `ReferenceCountedAsyncDisposable<T>` class, which is arguably a better match considering `IAsyncDisposable` does not implement `IDisposable`.

Reply:
yup.  agreed.